hypervisor/arch/x86/profiling.c

```diff
@@ -111,6 +229,100 @@ static void profiling_pmi_handler(__unused unsigned int irq, __unused void *data
 	/* to be implemented */
 }
 
+/*
+ * Initialize sep state and enable PMU counters
+ */
+void profiling_start_pmu(void)
+{
+	uint16_t i;
+
+	dev_dbg(ACRN_DBG_PROFILING, "%s: entering", __func__);
+
+	if (in_pmu_profiling) {
+		return;
+	}
+
+	for (i = 0; i < phys_cpu_num; i++) {
+		if (per_cpu(profiling_info.sep_state, i).pmu_state != PMU_SETUP) {
+			pr_err("%s: invalid pmu_state %u on cpu%d",
+			__func__, get_cpu_var(profiling_info.sep_state).pmu_state, i);
+			return;
+		}
+	}
+
+	for (i = 0; i < phys_cpu_num; i++) {
+		per_cpu(profiling_info.ipi_cmd, i) = IPI_PMU_START;
+		per_cpu(profiling_info.sep_state, i).samples_logged = 0U;
+		per_cpu(profiling_info.sep_state, i).samples_dropped = 0U;
+		per_cpu(profiling_info.sep_state, i).valid_pmi_count = 0U;
+		per_cpu(profiling_info.sep_state, i).total_pmi_count = 0U;
+		per_cpu(profiling_info.sep_state, i).total_vmexit_count = 0U;
+		per_cpu(profiling_info.sep_state, i).frozen_well = 0U;
+		per_cpu(profiling_info.sep_state, i).frozen_delayed = 0U;
+		per_cpu(profiling_info.sep_state, i).nofrozen_pmi = 0U;
+		per_cpu(profiling_info.sep_state, i).pmu_state = PMU_RUNNING;
+	}
+
+	smp_call_function(0xFUL, profiling_ipi_handler, NULL);
```

Comment:
do we need a lock to protect profiling_info.ipi_cmd?

Reply:
@anthonyzxu  ipi_cmd is specific to sep profiling and is maintained per cpu , and there is no other code that triggers change in that variable. So I think there is no need of lock.